test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java

```diff
@@ -99,16 +105,23 @@ void transformerBuildItem(BuildProducer<BytecodeTransformerBuildItem> transforme
             Set<String> sources = new HashSet<>();
             MultiSourceFileLocator sourceFileLocator = new MultiSourceFileLocator(4);
             if (BuildToolHelper.isMavenProject(targetdir.toPath())) {
+                Set<AppArtifactKey> runtimeDeps = new HashSet<>();
+                for (AppDependency i : curateOutcomeBuildItem.getEffectiveModel().getUserDependencies()) {
+                    runtimeDeps.add(i.getArtifact().getKey());
+                }
                 LocalProject project = LocalProject.loadWorkspace(targetdir.toPath());
+                runtimeDeps.add(project.getKey());
```

Comment:
@aloubyansky do you know if the gradle code below would also need a similar change? I don't have any multi module gradle projects locally to test on.

Reply:
Haven't tried it but looking at the change, it makes sense. BTW, the AppModel already includes `getLocalProjectArtifacts()` method, it could simplify your loops here.